autofs-5.1.9 - fix always recreate credential cache

From: Ian Kent <raven@themaw.net>

When I aplied the original patch from Ian Collier for this I changed
the credential end time comparison to be against the time returned from
monotomic_time(). But this isn't the same as the calander time returned
from time() which Ian used in his original patch.

Signed-off-by: Ian Kent < raven@themaw.net>
---
 CHANGELOG            |    1 +
 modules/cyrus-sasl.c |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG b/CHANGELOG
index fd9d8614a..3fe6b6310 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,5 +1,6 @@
 
 - Fix incompatible function pointer types in cyrus-sasl module.
+- fix always recreate credential cache.
 
 02/11/2023 autofs-5.1.9
 - fix kernel mount status notification.
diff --git a/modules/cyrus-sasl.c b/modules/cyrus-sasl.c
index 78b77942b..69e613bad 100644
--- a/modules/cyrus-sasl.c
+++ b/modules/cyrus-sasl.c
@@ -676,7 +676,7 @@ sasl_do_kinit(unsigned logopt, struct lookup_context *ctxt)
 	}
 	else {
 		krb5_creds match_creds, out_creds;
-		time_t now = monotonic_time(NULL);
+		time_t now = time(NULL);
 
 		/* even if the cache is in use, we will clear it if it
 		 * contains an expired credential for our principal,
